devices/interfaces.go

```diff
@@ -133,3 +137,9 @@ type Configure interface {
 	GenerateCSR(*cfgresources.HTTPSCertAttributes) ([]byte, error)
 	UploadHTTPSCert([]byte, string, []byte, string) (bool, error)
 }
+
+type Firmware interface {
```

Comment:
we should maybe have a longer conversation here, but I'm not too keen on adding another duplicate interface here when we've defined the same ones in `bmc/firmware.go`. The general direction I thought we were taking was to deprecate all the interfaces from this file in favor of the `bmc` directory interfaces. Thoughts?

Reply:
I do agree the newer `bmc` directory interface is the way ahead and that is what we will use. I've gone ahead and completed the implementation in the newer `bmc` directory interface. I'm okay with purging the added `Firmware` interface once we've discussed this a bit. 

For the backstory - I started off implementing the newer interface and realised a few things and went back to the older interface - I have outlined those points in https://github.com/bmc-toolbox/bmclib/issues/199 where we could continue the discussion.

Reply:
Purged in fca4f2e66bec5d2a54703ea57d50b8211a0cc1d0 the Firmware methods are now only available from the v2 interface

Reply:
fixed in 09da9d44d